src/main/java/zone/rong/loliasm/client/sprite/FramesTextureData.java

```diff
@@ -3,69 +3,118 @@
 import net.minecraft.client.Minecraft;
 import net.minecraft.client.renderer.texture.TextureAtlasSprite;
 import net.minecraft.client.renderer.texture.TextureMap;
+import net.minecraft.client.resources.IReloadableResourceManager;
 import net.minecraft.client.resources.IResource;
 import net.minecraft.client.resources.IResourceManager;
 import net.minecraft.util.ResourceLocation;
+import net.minecraftforge.client.event.ColorHandlerEvent;
+import net.minecraftforge.client.resource.ISelectiveResourceReloadListener;
+import net.minecraftforge.client.resource.VanillaResourceType;
 import net.minecraftforge.fml.common.eventhandler.SubscribeEvent;
 import net.minecraftforge.fml.common.gameevent.TickEvent;
 import zone.rong.loliasm.LoliLogger;
+import zone.rong.loliasm.LoliReflector;
+import zone.rong.loliasm.common.internal.mixins.TextureAtlasSpriteAccessor;
+import zone.rong.loliasm.common.internal.mixins.TextureMapAccessor;
 import zone.rong.loliasm.proxy.ClientProxy;
 
 import java.io.IOException;
 import java.util.*;
+import java.util.stream.Collectors;
 
 public class FramesTextureData extends ArrayList<int[][]> {
+    private static boolean canReload = true;
 
-    private static final Set<TextureAtlasSprite> scheduledToReleaseCache = Collections.newSetFromMap(new WeakHashMap<>());
+    private static final Class<?> FOAMFIX_SPRITE;
+
+    static {
+        Class<?> ffSprite;
+        try {
+            ffSprite = Class.forName("pl.asie.foamfix.client.FastTextureAtlasSprite");
+        } catch(ClassNotFoundException e) {
+            ffSprite = null;
+        }
+        FOAMFIX_SPRITE = ffSprite;
+    }
+
+    @SubscribeEvent
+    public static void registerEvictionListener(ColorHandlerEvent.Block event) {
```

Comment:
Using `ColorHandlerEvent.Block` here because its straight after models are baked? Or is there a different reason?

Reply:
Yeah, strategically chosen to inject the resource reload listener at the right point in the list.